### PR TITLE
let dinifile.readFromEnv()'s environment parameter be const

### PR DESCRIPTION
**Pull request by Herringway**
More const is nice, yes?

### CODE REVIEW

**Review by RazvanN7 — APPROVED** [review]

**Comment by RazvanN7** [comment]
Buildkite failure is unrelated.

**Review by jacob-carlborg — APPROVED** [review]

**Comment by wilzbach** [comment]
CC @CyberShadow. Did you change anything in are recently? It's failing now on Buildkite due to a missing file.

**Comment by CyberShadow** [comment]
> Did you change anything in are recently?

Yes, I did. I changed the structure of dub.sdl and updated `.travis.yml` too. But, the tests pass on my machine and on Travis. It seems to be a problem with the script running the tests on BuildKite - maybe it doesn't know how to test subpackages?

**Comment by CyberShadow** [comment]
The problem is here:

https://github.com/dlang/ci/blob/master/buildkite/build_project.sh#L232-L233

**Comment by CyberShadow** [comment]
I already excluded the previous Jenkins-based project tester by checking the environment:

https://github.com/cybershadow/ae/blob/master/sys/net/test.d#L76-L79

I've updated it to check for BuildKite:

https://github.com/CyberShadow/ae/releases/tag/v0.0.2389

Hopefully that will work.

File: src/dmd/dinifile.d [hunk 1]
```diff
@@ -113,12 +113,12 @@ const(char)[] findConfFile(const(char)[] argv0, const(char)[] inifile)
...
  *      environment value corresponding to name
  */
-const(char)* readFromEnv(StringTable* environment, const(char)* name)
+const(char)* readFromEnv(const(StringTable)* environment, const(char)* name)
 {
     const len = strlen(name);
-    auto sv = environment.lookup(name, len);
+    const sv = environment.lookup(name, len);
     if (sv && sv.ptrvalue)
-        return cast(const(char)*)sv.ptrvalue; // get cached value
```
Comment (Geod24) [inline src/dmd/dinifile.d]: Why was this `cast` removed ?

**Pull request merged**